### ISSUE TITLE
Colored text

### ISSUE THREAD

**Issue by jroimartin** [jroimartin/gocui#9]
Add support for colored text. E.g.: fmt.Println(v, "Regular text and [yb:Yellow bold text]")

**Comment by beefsack** [comment]
Hi @jroimartin, just wondering if there was any plan around this one?  Would love to use this library over using termbox directly for a project I'm just kicking off.

Happy to contribute if you had a high level idea of where you want to go with it.

**Comment by jroimartin** [comment]
Hi Michael,

Right now I'm working on giving gocui a real full edition mode. And then I
would like to decouple it, so the user can set his own keybinding. This way
it would be trivial to set up an emacs mode, vim mode or custom mode.

After finishing these two tasks my idea is to implement color support, but
first I think we should discuss about which is the better approach for
defining colors. E.g.: using ANSI codes, user-friendly syntax, etc. Also
it  will probably change some internals...

In short, my plan is to implement it after finishing the full edition mode
(very soon I hope). But if you want to discuss about it and contribute with
a PR that would be awesome! :)

**Comment by beefsack** [comment]
Hi @jroimartin, the new version sounds good, might be worth waiting until that's done.

The benefit of ANSI codes in the input might be because you could still use an `io.Writer` without big changes to the API.  This would actually work quite well for my project as I've already got an ANSI renderer in place.

The benefit to having a concrete API for triggering colour changes would be user friendliness.

It would be interesting to see if it would work having a defined interface, but internally it just generated the appropriate ANSI escape code and sent it to the `io.Writer`.  This might mean that you could have the best of both worlds.

**Comment by hojgr** [comment]
Hello,

I would like to know what's the status for this? I would like to use colored text in a project I am working on. I tried ANSI but those are apparently escaped :)

Thanks

**Comment by jroimartin** [comment]
It is still a work in progress and, hopefully, I will implement it soon using ANSI codes to specify the colours.

**Comment by deweerdt** [comment]
I've implemented basic support for ANSI escape characters in https://github.com/jroimartin/gocui/pull/39

**Comment by jroimartin** [comment]
@deweerdt cool! Color support is the number 1 priority for v0.3. I'll give it a look in the next days :)

**Comment by deweerdt** [comment]
@jroimartin I've noticed some issues yesterday with the current patch: attributes don't survive  a line change. I'll post an updated version this week.

**Comment by deweerdt** [comment]
Ok, I've added the fix to PR #39

**Comment by jroimartin** [comment]
Merged!
